packages/language-server/src/__test__/completion.test.ts

```diff
@@ -1190,6 +1252,303 @@ suite('Completions', function () {
+        },
+      })
+    })
+    test('MongoDB: @@index(address[|])', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([address|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'id', kind: CompletionItemKind.Field },
+            { label: 'email', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
+    test('MongoDB: @@index(address.[|]) first position, with only one type', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([address.|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'street', kind: CompletionItemKind.Field },
+            { label: 'number', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
+    test('MongoDB: @@index(address.[|]) with composite type suggestion 1', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+            alpha  Alpha
+          }
+          type Alpha {
+            bravo  Bravo
+            helloA Int
+          }
+          type Bravo {
+            something  String
+            helloBravo Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([address.|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [
+            { label: 'street', kind: CompletionItemKind.Field },
+            { label: 'number', kind: CompletionItemKind.Field },
+            { label: 'alpha', kind: CompletionItemKind.Field },
+          ],
+        },
+      })
+    })
+    test('MongoDB: @@index(address.a[|]) with composite type suggestion 1', () => {
+      assertCompletion({
+        provider: 'mongodb',
+        schema: /* Prisma */ `
+          type Address {
+            street String
+            number Int
+            alpha  Alpha
+          }
+          type Alpha {
+            bravo  Bravo
+            helloA Int
+          }
+          type Bravo {
+            something  String
+            helloBravo Int
+          }
+          model User {
+            id      Int     @id
+            email   String
+            address Address
+            @@index([address.a|])
+          }`,
+        expected: {
+          isIncomplete: false,
+          items: [],
```

Comment:
Should this actually suggest `alpha` here?

Reply:
added a TODO comment